File: 80_remove_duplicates_from_sorted_array_II.cxx
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int cnt=0;
        for(auto i=++nums.begin(); i!=nums.end(); ) {
            if(*i == *(i-1)) {
                ++cnt>1 ? nums.erase(i) : ++i;
            } else {
                cnt=0;
                ++i;
            }
        }
        return nums.size();
    }
};

File: 80_remove_duplicates_from_sorted_array_II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        cnt=0
        i=1
        while i<len(nums):
            if nums[i]==nums[i-1]:
                cnt+=1
                if cnt>1: 
                    nums.pop(i)
                else:
                    i+=1
            else:
                cnt=0
                i+=1
        return len(nums)
